### PR TITLE
Migrate genesis.yml from block_version to consensus

### PR DESCRIPTION
**Pull request by CodeSandwich**
Finish block and consensus versioning split. Switch genesis.yml from pretending that it accepts block version to accepting consensus version without changing meaning of accepted values.

### CODE REVIEW

File: doc/genesis_file.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ initial_setting:
   bootstrap_key_slots_percentage: ~
   slot_duration: 15
   epoch_stability_depth: 10
-  block_version: 1
+  consensus: 1
```
Comment (NicolasDP) [inline doc/genesis_file.md]: actually since you are there, can you put a string here?

`bft` or `genesis`

Comment (vincenthz) [reply doc/genesis_file.md]: here and also in the block0 state

Comment (NicolasDP) [inline doc/genesis_file.md]: I think this should be `bft` and  no longer `1`right ?
```suggestion
  consensus: bft
```

Comment (CodeSandwich) [reply doc/genesis_file.md]: @vincenthz I may be looking at the wrong place, but AFAIK block0 is `chain_impl_mockchain::block::Block`, which is either stored as binary or generated from genesis YAML in CLI. Using string representation in binary is not a good idea and genesis.yml is already switched to using string. What am I missing?

Comment (vincenthz) [reply doc/genesis_file.md]: I'm talking about the Message `Initial` and its content related to ConfigParam in mockchain config.rs

Comment (CodeSandwich) [reply doc/genesis_file.md]: `Message::Initial` contains tag-value pairs and the only tags we support for now are `Block0Date` (1) and `Discriminant` (2). I don't see state, block version or consensus version. Should I add another tag for consensus version with string value?

Comment (vincenthz) [reply doc/genesis_file.md]: yes, we are indeed missing consensus as one parameter. this is crucial to know what consensus the blockchain is starting with (which is what the `Initial` message is about)

Comment (CodeSandwich) [reply doc/genesis_file.md]: Ok, I'll add it. Should it go in this PR or the next one?

Comment (vincenthz) [reply doc/genesis_file.md]: it can be the next one

**Pull request merged**